apps/web/src/pages/integrations/UpdateProviderPage.tsx

```diff
@@ -1,42 +1,329 @@
-import { useState } from 'react';
+import { useEffect, useState } from 'react';
 import styled from '@emotion/styled';
-import { useMantineColorScheme } from '@mantine/core';
-import { colors } from '../../design-system';
+import { ActionIcon, ColorScheme, Group, TextInput, useMantineColorScheme, Text, Loader } from '@mantine/core';
+import { Button, colors, Input, Switch } from '../../design-system';
 import { useNavigate, useParams } from 'react-router-dom';
-import { useEffectOnce } from '../../hooks';
 import { useProviders } from './useProviders';
 import { IIntegratedProvider } from './IntegrationsStoreModal';
+import { Check, Close, Copy } from '../../design-system/icons';
+import { Controller, useForm } from 'react-hook-form';
+import { CONTEXT_PATH } from '../../config';
+import { IConfigCredentials } from '@novu/shared';
+import { IntegrationInput } from './components/IntegrationInput';
+import { useClipboard } from '@mantine/hooks';
+import slugify from 'slugify';
+import { IntegrationChannel } from './components/IntegrationChannel';
+import { CHANNEL_TYPE_TO_STRING } from '../../utils/channels';
+import { IntegrationEnvironmentPill } from './components/IntegrationEnvironmentPill';
+import { useFetchEnvironments } from '../../hooks/useFetchEnvironments';
+
+const getLogoFileName = (id, schema: ColorScheme) => {
+  if (schema === 'dark') {
+    return `${CONTEXT_PATH}/static/images/providers/dark/square/${id}.svg`;
+  }
+
+  return `${CONTEXT_PATH}/static/images/providers/light/square/${id}.svg`;
+};
+
+interface IProviderForm {
+  name: string;
+  credentials: Record<string, string>;
+  active: boolean;
+  identifier: string;
+}
 
 export function UpdateProviderPage() {
+  const { environments, isLoading: areEnvironmentsLoading } = useFetchEnvironments();
   const [selectedProvider, setSelectedProvider] = useState<IIntegratedProvider | null>(null);
-  const { providers } = useProviders();
+  const { providers, refetch, isLoading } = useProviders();
   const { colorScheme } = useMantineColorScheme();
   const isDark = colorScheme === 'dark';
   const { integrationId } = useParams();
   const navigate = useNavigate();
 
-  useEffectOnce(() => {
+  const {
+    control,
+    handleSubmit,
+    reset,
+    watch,
+    setValue,
+    formState: { errors, isDirty },
+  } = useForm<IProviderForm>({
+    shouldUseNativeValidation: false,
+    defaultValues: {
+      name: '',
+      credentials: {},
+      active: false,
+      identifier: '',
+    },
+  });
+
+  const isActive = watch('active');
+  const name = watch('name');
+  const identifier = watch('identifier');
+  const identifierClipboard = useClipboard({ timeout: 1000 });
+
+  useEffect(() => {
```

Comment:
Need to look at this when we have create/update api in place maybe this is not needed?